subprojects/docs/src/docs/userguide/custom_plugins.adoc

```diff
@@ -40,100 +40,143 @@ You can create a separate project for your plugin. This project produces and pub
 
 The following demonstrates the usage of the type for mapping an extension property to a task property:
 
-++++
-<sample id="mapExtensionPropertiesToTaskProperties" dir="userguide/tasks/mapExtensionPropertiesToTaskProperties" includeLocation="true" title="Mapping extension properties to task properties">
-            <sourcefile file="build.gradle"/>
-            <output args="-q hello"/>
-        </sample>
-++++
+=== Example: Mapping extension properties to task properties
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/userguide/tasks/mapExtensionPropertiesToTaskProperties/build.gradle[]
+----
+
+NOTE: The code for this example can be found at **`samples/userguide/tasks/mapExtensionPropertiesToTaskProperties`** in the ‘-all’ distribution of Gradle.
+
+.Output of `gradle -q hello`
+----
+> gradle -q hello
+include::{samplesPath}/userguide/tasks/mapExtensionPropertiesToTaskProperties/mapExtensionPropertiesToTaskProperties.out[]
+----
 
 [[sec:custom_plugins_standalone_project]]
-=== A standalone project
+== A standalone project
 
 Now we will move our plugin to a standalone project, so we can publish it and share it with others. This project is simply a Groovy project that produces a JAR containing the plugin classes. Here is a simple build script for the project. It applies the Groovy plugin, and adds the Gradle API as a compile-time dependency.
 
-++++
-<sample id="customPluginStandalone" dir="customPlugin/plugin" title="A build for a custom plugin" includeLocation="true">
-            <sourcefile file="build.gradle" snippet="use-plugin"/>
-        </sample>
-++++
+=== Example: A build for a custom plugin
 
-So how does Gradle find the api:org.gradle.api.Plugin[] implementation? The answer is you need to provide a properties file in the jar's `META-INF/gradle-plugins` directory that matches the id of your plugin.
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/customPlugin/plugin/build.gradle[tag=use-plugin]
+----
 
-++++
-<sample id="customPluginStandalone" dir="customPlugin/plugin" title="Wiring for a custom plugin">
-            <sourcefile file="src/main/resources/META-INF/gradle-plugins/org.samples.greeting.properties"/>
-        </sample>
-++++
+NOTE: The code for this example can be found at **`samples/customPlugin/plugin`** in the ‘-all’ distribution of Gradle.
 
-Notice that the properties filename matches the plugin id and is placed in the resources folder, and that the `implementation-class` property identifies the api:org.gradle.api.Plugin[] implementation class.
+So how does Gradle find the link:{javadocPath}/org/gradle/api/Plugin.html[Plugin] implementation? The answer is you need to provide a properties file in the jar's `META-INF/gradle-plugins` directory that matches the id of your plugin.
+
+=== Example: Wiring for a custom plugin
+
+[source.multi-language-sample,groovy]
```

Comment:
:x: This should be a `[source,properties]` or just `[source]` if `properties` highlighting isn't supported